docs/connectors/sinks/delta.md

```diff
@@ -0,0 +1,125 @@
+  * `append` - New updates will be appended to the existing table at the target location
+  * `truncate` - Existing table at the specified location will get truncated. The connector
+     truncates the table by outputing delete actions for all files in the latest snapshot
+     of the table.
+  * `error_if_exists` - If a table exists at the specified location, the operation will fail.
+
+### Storage parameters
+
+Additional configuration options are defined for specific storage backends.  Refer to
+backend-specific documentation for details:
+
+* [Amazon S3 options](https://docs.rs/object_store/latest/object_store/aws/enum.AmazonS3ConfigKey.html)
+* [Azure Blob Storage options](https://docs.rs/object_store/latest/object_store/azure/enum.AzureConfigKey.html)
+* [Google Cloud Storage options](https://docs.rs/object_store/latest/object_store/gcp/enum.GoogleConfigKey.html)
+
+## The small file problem and output buffer configuration
+
+By default a Feldera pipeline sends a batch of changes to the output transport
+for each batch of input updates it processes.  This can result in a stream of
+small updates, which is normal and even preferable for output transports like
+Kafka; however it can cause problems for the Delta Lake format by creating a large
+number of small files.
+
+The output buffer mechanism is designed to solve this problem by decoupling the
+rate at which the pipeline pushes changes to the output transport from the rate
+of input changes.  It works by accumulating updates inside the pipeline
+for up to a user-defined period of time or until accumulating a user-defined number
+of updates and writing them to the Delta Table as a small number of large files.
+
+The following parameters are used to configure the output buffer
+
+* `enable_output_buffer` - Enable output buffer.
+
+* `max_output_buffer_time_millis` - Maximum time in milliseconds data is kept
+   in the output buffer.
+
+   When not specified, data is kept in the buffer indefinitely until one of
+   the other trigger conditions is satisfied.  When this option is
+   set the buffer will be flushed at most every
+   `max_output_buffer_time_millis` milliseconds.
+
+   NOTE: this configuration option requires the `enable_output_buffer` flag
+   to be set.
+
+* `max_output_buffer_size_records` - Maximum number of updates to be kept in
+   the output buffer.
+
+   This parameter bounds the maximal size of the buffer.  
+   Note that the size of the buffer is not always equal to the
+   total number of updates output by the pipeline. Updates to the
+   same record can overwrite or cancel previous updates.
+
+   When not specified, the buffer can grow indefinitely until one of
+   the other trigger conditions is satisfied.
+
+   NOTE: this configuration option requires the `enable_output_buffer` flag
+   to be set.
+
+:::note
+
+when the `enable_output_buffer` flag is set, at least one of
+`max_output_buffer_time_millis` or `max_output_buffer_size_records` must be
```

Comment:
Would this misconfiguration return a useful error to the user?

Reply:
It shoud now.